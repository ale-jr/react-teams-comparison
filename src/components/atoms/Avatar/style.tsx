import styled from "styled-components";
import { SizingKeys, Sizing } from 'interfaces/theme'

export interface AvatarProps {
    size: SizingKeys,
}

const avatarSizes: Sizing = {
    xs: '1em',
    sm: '1.5em',
    md: '2em',
    lg: '3em',
    xl: '4em'
}



const Avatar = styled.img<AvatarProps>`
    //Position and Layout


    //Display and visibility


    //Clipping


    //Animation

    
    //Box model (From outside in)
    width: ${({ size }) => avatarSizes[size]};
    height: ${({ size }) => avatarSizes[size]};

    //Background


    //Typography


    //Pseudo-classes and pseudo-elements

`

export default Avatar