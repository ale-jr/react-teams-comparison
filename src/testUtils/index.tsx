import { render, RenderOptions, RenderResult } from '@testing-library/react'
import ThemeProvider from 'providers/ThemeProvider'
import { ReactElement } from 'react'


const Providers: React.FC<{
    children: JSX.Element | JSX.Element[]
}> = ({ children }) => <ThemeProvider>{children}</ThemeProvider>


const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "queries">): RenderResult => render(ui, {
    wrapper: Providers,
    ...options
})

export * from '@testing-library/react'
export {customRender as render}