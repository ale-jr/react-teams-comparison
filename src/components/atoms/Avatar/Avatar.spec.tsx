import { render } from 'testUtils'
import 'jest-styled-components'
import Avatar from './index'


describe("Avatar", () => {
    it("Should match snapshot", () => {
        const { asFragment } = render(<Avatar src="/test.png" size='md' />)
        expect(asFragment()).toMatchSnapshot()
    })
})