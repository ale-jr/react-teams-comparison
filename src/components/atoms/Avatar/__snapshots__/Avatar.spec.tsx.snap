// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  width: 2em;
  height: 2em;
}

<img
    class="c0"
    src="/test.png"
  />
</DocumentFragment>
`;
